cleanrl/atari/v_c51_atari.py

```diff
@@ -0,0 +1,283 @@
+            nn.Conv2d(frames, 32, 8, stride=4),
+            nn.ReLU(),
+            nn.Conv2d(32, 64, 4, stride=2),
+            nn.ReLU(),
+            nn.Conv2d(64, 64, 3, stride=1),
+            nn.ReLU(),
+            nn.Flatten(),
+            nn.Linear(3136, 512),
+            nn.ReLU(),
+            nn.Linear(512, envs.single_action_space.n * n_atoms)
+        )
+
+    def forward(self, x, device):
+        return self.network(x/255.0)
+
+    def get_action(self, envs, device, x,action=None):
+        logits = self.forward(x, device)
+        # probability mass function for each action
+        pmfs = torch.softmax(logits.view(len(x), envs.single_action_space.n, self.n_atoms), dim=2)
+        q_values = (pmfs*self.atoms).sum(2)
+        if action is None:
+            action = torch.argmax(q_values, 1)
+            random_action = torch.randint(0, envs.single_action_space.n, (len(x),), device=device)
+            random_action_flag = torch.rand(len(x), device=device) > epsilon
+            action = torch.where(random_action_flag, action, random_action)
+           
+        return action, pmfs[torch.arange(len(x)), action]
+
+def linear_schedule(start_e: float, end_e: float, duration: int, t: int):
+    slope =  (end_e - start_e) / duration
+    return max(slope * t + start_e, end_e)
+
+if __name__ == "__main__":
+    args = parse_args()
+    run_name = f"{args.gym_id}__{args.exp_name}__{args.seed}__{int(time.time())}"
+    writer = SummaryWriter(f"runs/{run_name}")
+    writer.add_text('hyperparameters', "|param|value|\n|-|-|\n%s" % (
+            '\n'.join([f"|{key}|{value}|" for key, value in vars(args).items()])))
+    if args.track:
+        import wandb
+        wandb.init(project=args.wandb_project_name, 
+                   entity=args.wandb_entity,
+                   sync_tensorboard=True,
+                   config=vars(args),
+                   name=run_name,
+                   monitor_gym=True,
+                   save_code=True)
+
+    # TRY NOT TO MODIFY: seeding
+    random.seed(args.seed)
+    np.random.seed(args.seed)
+    torch.manual_seed(args.seed)
+    torch.backends.cudnn.deterministic = args.torch_deterministic
+    
+    device = torch.device('cuda' if torch.cuda.is_available() and args.cuda else 'cpu')
+
+    # env setup
+    envs = gym.vector.SyncVectorEnv(
+        [make_env(args.gym_id, args.seed + i, i, args.capture_video, run_name) for i in range(args.num_envs)]
+    )
+    assert isinstance(envs.single_action_space, gym.spaces.Discrete), "only discrete action space is supported"
+
+    # ALGO LOGIC: initialize agent here:
+    q_network = QNetwork(envs, device, n_atoms=args.n_atoms, v_min=args.v_min, v_max=args.v_max).to(device)
+    target_network = QNetwork(envs, device, n_atoms=args.n_atoms, v_min=args.v_min, v_max=args.v_max).to(device)
+    target_network.load_state_dict(q_network.state_dict())
+    optimizer = optim.Adam(q_network.parameters(), lr=args.learning_rate, eps=0.01/args.batch_size)

```

Comment:
is modifying `eps` necessary?

Reply:
I got it from c51_atari.py file. Can remove it if required

https://github.com/vwxyzjn/cleanrl/blob/d1002bd80cecda4e7251460a1db1c832943578e8/cleanrl/c51_atari.py#L491

Reply:
Oh interesting. I forgot about this. Then keep it as it is please